packages/datastore/datastore/src/storage/adapters/indexedDB/IndexedDBStorageAdapter.ts

```diff
@@ -115,13 +115,31 @@ export class IndexedDBStorageAdapter implements StorageAdapter {
     return this.convertToPromise<any>(store.get(key));
   }
 
-  public async query(storeName: string, filter?: Filter) {
+  public async query(storeName: string, filter?: Filter): Promise<any[]> {
     const store = await this.getStore(storeName);
-    const all = await this.convertToPromise<any[]>(store.getAll());
+    const cursorReq = store.openCursor();
+    const result: any[] = [];
+
+    return new Promise((resolve, reject) => {
+      cursorReq.onsuccess = () => {
+        const cursor = cursorReq.result;
+        if (cursor) {
+          if (!filter) {
+            result.push(cursor.value);
+          } else {
+            const predicate = getPredicate(filter);
+            if (predicate(cursor.value)) {
```

Comment:
Why we have done this? Any reason to use cursor vs full data read?

Reply:
Any issue where we discussed that? Want to get more context

Reply:
I don't think we discussed this in an issue be we have discussed this on the discord chat. Reason for Cursors is to avoid loading all data into memory. Without Cursors, we have to load all data into memory before filtering. Although, If there is no filter, we don't need Cursors since we still load all data. But when there is a filter, we only keep results in memory.

Reply:
I do not like us adding some optimatizations like results in memory. Performance etc. Without actually having performance tests and general idea how this works. Let's keep it but please do not do any optimalizations without discussion